components/signoffs/signoffController.js

```diff
@@ -0,0 +1,170 @@
+'use strict';
+
+const signoffDao = require('./signoffDao');
+const JSONStream = require('JSONStream');
+const {
+  handleErrors,
+  ValidationError,
+  ConflictingDataError,
+} = require('../../utils/controllerUtils');
+const { signoff } = require('../../utils/urlSlugs');
+const logger = require('../../utils/logger').logger;
+
+const getSignoff = async (req, res, next) => {
+  try {
+    const signoffs = await signoffDao.getByCaseIdentifier(
+      req.params.caseIdentifier
+    );
+    if (signoffs) {
+      res.status(200).json(signoffs);
+    } else {
+      res.status(404).end();
+    }
+  } catch (e) {
+    handleErrors(e, 'Error getting signoff(s) for case', logger, next);
+  }
+};
+
+const addSignoff = async (req, res, next) => {
+  try {
+    const existingSignoff = null;
+    //const existingSignoff = await signoffDao.getByCaseIdentifier(
+    //  req.body.caseIdentifier
+    //);
+    //const validationResults = validateObjectsFromUser(req.body);
+    if (existingSignoff == null) {
+      await upsert(req.body);
+      res.status(201).end();
+    } else if (
+      existingSignoff.username == req.body.username &&
+      existingSignoff.signoff_step_name == req.body.signoff_step_name &&
+      existingSignoff.deliverable_type == req.body.deliverable_type
+    ) {
+      // signoff step is same, replace the old signoff record
+      await upsert(req.body);
+      res.status(201).end();
+    } else {
+      // signoff data is different
+      await upsert(req.body);
+      res.status(201).end();
```

Comment:
It looks like this logic is the same logic as for if the signoff already exists - does it need a separate branch?